sftp.go

```diff
@@ -227,3 +231,25 @@ func (s *StatusError) Error() string {
 func (s *StatusError) FxCode() fxerr {
 	return fxerr(s.Code)
 }
+
+func isSFTPExtensionSupported(extension SSHExtensionPair) bool {
+	for _, supportedExtension := range supportedSFTPExtensions {
+		if supportedExtension == extension {
+			return true
+		}
+	}
+	return false
+}
+
+// SetSFTPExtensions allows to customize the supported server extensions.
+// See the variable supportedSFTPExtensions for supported extensions.
+// If an invalid extension is given an error will be returned
+func SetSFTPExtensions(extensions []SSHExtensionPair) error {
+	for _, extension := range extensions {
+		if !isSFTPExtensionSupported(extension) {
+			return fmt.Errorf("Unsupported extension, name: %v data: %v", extension.Name, extension.Data)
+		}
+	}
+	sftpExtensions = extensions
```

Comment:
😮 I just realized something bad could happen here.

Please include this test in your PR, and then fix your code so that it passes:
```
func TestSetSFTPExtensions_IsNotMutable(t *testing.T) {
	inputExtensions := []SSHExtensionPair{
		{"hardlink@openssh.com", "1"},
	}

	// It is important that this be a separate slice, as we mutate inputExtensions below.
	expectExtensions := []SSHExtensionPair{
		{"hardlink@openssh.com", "1"},
	}

	err := SetSFTPExtensions(inputExtensions)
	assert.NoError(t, err)
	assert.Equal(t, expectExtensions, sftpExtensions)

	inputExtensions[0].Data = 42
	assert.Equal(t, expectExtensions, sftpExtensions)
}
```

Reply:
should be fixed, thanks!